lib/phoenix_live_view/diff.ex

```diff
@@ -343,4 +383,13 @@ defmodule Phoenix.LiveView.Diff do
     id_to_components = Map.put(id_to_components, id, dump_component(socket, cid))
     {pending_components, component_diffs, {id_to_components, cid_to_ids, uuids}}
   end
+
+  defp fetch_component!(module, id, components) do
+    {id_to_components, _cid_to_ids, _} = components
+    case Map.fetch(id_to_components, {module, id}) do
+      {:ok, {cid, assigns, _, _}} -> {cid, assigns}
+      :error ->
+        raise ArgumentError, "no #{inspect(module)} component found with :id #{inspect(id)}"
```

Comment:
I think we should just ignore missing components. For example, between a `send_update`, the browser could have removed the ID. Another option is to log (warn or info).

Reply:
Also don't forget to run the fomatter. :D